provider/schema/execution.go

```diff
@@ -316,7 +335,28 @@ func (e *ExecutionData) resolveResourceValues(ctx context.Context, meta ClientMe
 			err = fmt.Errorf("recovered from panic: %s", r)
 		}
 	}()
-	if err = e.resolveColumns(ctx, meta, resource, resource.table.Columns); err != nil {
+
+	cols := e.Db.Dialect().Columns(resource.table)
+
+	providerCols, internalCols := make([]Column, 0, len(cols)), make([]Column, 0, len(cols))
+	cqIdColIndex := -1
+	for i := range cols {
+		if cols[i].internal {
+			if cols[i].Name == cqIdColumn.Name {
+				cqIdColIndex = len(internalCols)
+			}
+
+			internalCols = append(internalCols, cols[i])
+		} else {
+			providerCols = append(providerCols, cols[i])
+		}
+	}
+	// resolve cqId last, as it would need other PKs to be resolved, some might be internal (cq_fetch_date)
+	if lastIndex := len(internalCols) - 1; cqIdColIndex > -1 && cqIdColIndex != lastIndex {
+		internalCols[cqIdColIndex], internalCols[lastIndex] = internalCols[lastIndex], internalCols[cqIdColIndex]
+	}
+
+	if err = e.resolveColumns(ctx, meta, resource, providerCols); err != nil {
```

Comment:
why not call resolve columns once on normal and then once again on all internal ones? and do a method that returns all columns and one that returns internal columns?

Reply:
Fixed. Why resolve multiple times when you can do just once?